cdap-app-fabric/src/main/java/co/cask/cdap/app/DefaultAppConfigurer.java

```diff
@@ -204,6 +205,10 @@ public void addSchedule(Schedule schedule, SchedulableProgramType programType, S
     Preconditions.checkArgument(!programName.isEmpty(), "Program name cannot be empty.");
     Preconditions.checkArgument(!schedules.containsKey(schedule.getName()), "Schedule with the name '" +
       schedule.getName()  + "' already exists.");
+    if (schedule instanceof StreamSizeSchedule) {
```

Comment:
If it is a Schedule (that is app.configure() used the deprecated Schedule() constructor directly), then you can replace it with a TimeSchedule here? That makes life easier in other places, because there you can expect that no plain Schedule will appear any more.